manifests/plugin/processes.pp

```diff
@@ -17,27 +17,33 @@
     interval => $interval,
   }
 
-  concat{"${collectd::params::plugin_conf_dir}/processes-config.conf":
-    ensure         => $ensure,
-    mode           => '0640',
-    owner          => 'root',
-    group          => $collectd::params::root_group,
-    notify         => Service['collectd'],
-    ensure_newline => true,
+  if ( $processes or $processes_matches ) {
+    concat{"${collectd::params::plugin_conf_dir}/processes-config.conf":
+      ensure         => $ensure,
```

Comment:
Would it not be simpler, in order to avoid this huge if block, to use an alternate `$ensure` value that is derived from the condition?

    if ( $processes or $processes_matches ) {
      $process_config_ensure = true
    } else { ...